### DISCUSSION TITLE
identify reports wrong bit depth for 32-bit image

### DISCUSSION THREAD

**Discussion by carterbox** [ImageMagick/ImageMagick#6731] [Contribute]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

ImageMagick does not recognize 32-bit images correctly (which leads to corruption of the images). Example files [available here](https://drive.google.com/open?id=1-3HosxGc-Gp2Cxh4ZNb6D8Wv21uO2A8N). By corrupted, I mean that images, that were once viewable using ImageJ, are now incorrectly formatted.

![Screenshot from 2020-04-27 12-37-18](https://user-images.githubusercontent.com/9604511/80402696-e4dcf280-8883-11ea-993b-f7d6d0857b3f.png)

### Steps to Reproduce

```bash
$ tiffinfo before-magick.tiff
TIFF Directory at offset 0x8 (8)
  Image Width: 1272 Image Length: 788
  Resolution: 1, 1 (unitless)
  Bits/Sample: 32   # this is the correct bit depth
  Sample Format: IEEE floating point
  Compression Scheme: None
  Photometric Interpretation: min-is-black
  Samples/Pixel: 1
  Rows/Strip: 788
  Planar Configuration: single image plane
  ImageDescription: {"shape": [788, 1272]}
  Software: tifffile.py

$ ./magick identify -verbose before-magick.tiff
Image: before-slack.tiff
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: DirectClass
  Geometry: 1272x788+0+0
  Resolution: 1x1
  Print size: 1272x788
  Units: Undefined
  Colorspace: Gray
  Type: Grayscale
  Endianness: LSB
  Depth: 32/16-bit
  Channel depth:
    Gray: 16-bit  # this bit depth is incorrect
  Channel statistics:
    Pixels: 1002336
    Gray:
      min: 0  (0)
      max: 65535 (1)
      mean: 7463.48 (0.113885)
      standard deviation: 19567.9 (0.298587)
      kurtosis: 4.12388
      skewness: 2.42943
      entropy: 0.153372
  ...  # lines removed for brevity
  Version: ImageMagick 7.0.10-9 Q16 x86_64 2020-04-26 https://imagemagick.org

$ convert before-magick.tiff after-magick.tiff
```
### System Configuration

ImageMagick 6.9.10-23 Q16 x86_64 2019010
ImageMagick 7.0.10-9 Q16 x86_64 2020-04-26
Ubuntu 20.04

**Comment by urban-warrior** [comment]
See https://imagemagick.org/script/formats.php.  Now try this command:

`magick before-magick.tiff -define quantum:format=floating-point after-magick.tiff`

Does that work for you?

**Comment by carterbox** [comment]
Thanks for the quick response. No, that doesn't work. I also tried:
`./magick before-magick.tiff -define quantum:format=floating-point -depth 32 after-magick.tiff`
